### PR TITLE
Fix/map ux interactions

### LINKED ISSUE #48: Highlight blocks even if they've been assigned to an existing zone
In the existing app, if you are hovering on selected zones the hover effect is still shown, and is darker / matching the tone of the selected zone.

image

We should do the same here.

### LINKED ISSUE #49: Don't show hover when pan tool is selected

### LINKED ISSUE #65: Toggle for brush size

### PR DESCRIPTION
**Pull request by mariogiampieri**
## Description

- closes #49 
- closes #48 
- closes #65 
- also fixes a bug where hover persists if you mouse off the layer / onto sidebar / etc. 

## Reviewers

- Primary: @raphaellaude 
- Secondary:

## Checklist

- [ ] Added/Updated related documentation (if applicable).
- [ ] Added/Updated related unit tests (if applicable).

### CODE REVIEW

File: app/src/app/utils/events/mapEvents.ts [hunk 1]
```diff
@@ -93,38 +92,47 @@ export const handleMapMouseDown = (
...
-  hoverFeatureIds: React.MutableRefObject<Set<string>>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
 ) => {
   const mapStore = useMapStore.getState();
   const activeTool = mapStore.activeTool;
   const isPainting = mapStore.isPainting;
   const sourceLayer = mapStore.selectedLayer?.name;
   const selectedFeatures = mapStore.paintFunction(map, e, mapStore.brushSize);
-  if (!isPainting && sourceLayer) {
+  if (!isPainting && sourceLayer && activeTool === "brush") {
```
Comment (raphaellaude) [inline app/src/app/utils/events/mapEvents.ts:135]: H: This is super close but I think the behavior we want is:
```javascript
if (sourceLayer && activeTool === "brush") {
    HighlightFeature(selectedFeatures, map, hoverFeatureIds, sourceLayer);
    if (isPainting) {
      // selects in the map object; the store object
      // is updated in the mouseup event
      SelectMapFeatures(selectedFeatures, map, mapStore);
    }
}
```
Right now, when you're painting the highlighted features don't follow your cursor, but rather are shown where the mouse down event event occurred. Instead, we should continue to highlight features under the cursor as you paint (tested locally).

**Review by raphaellaude — CHANGES_REQUESTED** [review]
Way to knock out four issues in one PR. Love it, esp. the minimal diff required to get this done. Only one blocking comment (and proposed + locally tested solution).

File: app/src/app/utils/events/mapEvents.ts [hunk 2]
```diff
@@ -28,7 +27,7 @@ MapEvent handling; these functions are called by the event listeners in the MapC
 export const handleMapClick = (
   e: MapLayerMouseEvent | MapLayerTouchEvent,
   map: MutableRefObject<Map | null>,
-  hoverFeatureIds: React.MutableRefObject<Set<string>>,
+  hoverFeatureIds: React.MutableRefObject<Set<string>>
```
[the remark this answers is not in the document]
Comment (mariogiampieri) [reply app/src/app/utils/events/mapEvents.ts:30]: i use and like `dbaeumer.vscode-eslint`

File: app/src/app/utils/events/mapEvents.ts (hunk 1 above)
Comment (mariogiampieri) [reply app/src/app/utils/events/mapEvents.ts:135]: good catch

File: app/src/app/utils/events/mapEvents.ts [hunk 3]
```diff
@@ -93,38 +92,47 @@ export const handleMapMouseDown = (
...
 ) => {
   const mapStore = useMapStore.getState();
   const activeTool = mapStore.activeTool;
   const isPainting = mapStore.isPainting;
   const sourceLayer = mapStore.selectedLayer?.name;
   const selectedFeatures = mapStore.paintFunction(map, e, mapStore.brushSize);
-  if (!isPainting && sourceLayer) {
+  if (sourceLayer && activeTool === "brush") {
     HighlightFeature(selectedFeatures, map, hoverFeatureIds, sourceLayer);
   } else if (activeTool === "brush" && isPainting && sourceLayer) {
```
Comment (raphaellaude) [inline app/src/app/utils/events/mapEvents.ts:137]: H: This now won't paint at all – the if statement needs to be within the first one or in an entirely separate conditional block.

> Use `else if` to specify a new condition to test, if the first condition is false

**Review by raphaellaude — CHANGES_REQUESTED** [review]

**Review by raphaellaude — APPROVED** [review]
lgtm

**Pull request merged**